commit 477e94221e14b2ab663bbf0ef09f443c03b86023
Author: Matthias Reichl <hias@horus.com>
Date:   Wed Jan 9 14:51:01 2019 +0100

    gpio-ir: change default pull configuration to up
    
    IR receivers like the TSOP series from Vishay and compatible ones
    have active-low open collector outputs with an internal pull up of
    about 30k (according to the TSOP datasheets).
    
    Activating a pull-down resistor on the GPIO will make it work against
    the pull-up in the IR receiver and brings the idle input voltage down
    to about 1.9V (measured on a RPi3B+ with a TSOP4438). While that's
    usually enough to make the RPi see a high signal it's certainly not
    optimal and may even fail when using an IR receiver with a weaker pull-up.
    
    Switching the default GPIO pull to "up" results in an input voltage
    level of about 3.3V and ensures that an idle state (high signal) will
    be detected if no IR receiver is attached.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 049b50535656..2872a00e1bcb 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -611,7 +611,7 @@ Load:   dtoverlay=gpio-ir,<param>=<val>
 Params: gpio_pin                Input pin number. Default is 18.
 
         gpio_pull               Desired pull-up/down state (off, down, up)
-                                Default is "down".
+                                Default is "up".
 
         rc-map-name             Default rc keymap (can also be changed by
                                 ir-keytable), defaults to "rc-rc6-mce"
diff --git a/arch/arm/boot/dts/overlays/gpio-ir-overlay.dts b/arch/arm/boot/dts/overlays/gpio-ir-overlay.dts
index 1bd9bb950efa..bd181236fcf2 100644
--- a/arch/arm/boot/dts/overlays/gpio-ir-overlay.dts
+++ b/arch/arm/boot/dts/overlays/gpio-ir-overlay.dts
@@ -30,7 +30,7 @@
                         gpio_ir_pins: gpio_ir_pins@12 {
                                 brcm,pins = <18>;                       // pin 18
                                 brcm,function = <0>;                    // in
-                                brcm,pull = <1>;                        // down
+                                brcm,pull = <2>;                        // up
                         };
                 };
         };
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1293_477e94221e14b2ab663bbf0ef09f443c03b86023.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1293_477e94221e14b2ab663bbf0ef09f443c03b86023.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
